### PR TITLE
Add EIP: Block-level Warming

### PR DESCRIPTION
**Pull request by forshtat**
**ATTENTION: ERC-RELATED PULL REQUESTS NOW OCCUR IN [ETHEREUM/ERCS](https://github.com/ethereum/ercs)**

--

When opening a pull request to submit a new EIP, please use the suggested template: https://github.com/ethereum/EIPs/blob/master/eip-template.md

We have a GitHub bot that automatically merges some PRs. It will merge yours immediately if certain criteria are met:

 - The PR edits only existing draft PRs.
 - The build passes.
 - Your GitHub username or email address is listed in the 'author' header of all affected PRs, inside <triangular brackets>.
 - If matching on email address, the email address is the one publicly listed on your GitHub profile.

### CODE REVIEW

File: EIPS/eip-7968.md [hunk 1]
```diff
@@ -0,0 +1,370 @@
+---
+eip: 7968
```
Comment (abcoathup) [inline EIPS/eip-7968.md:2]:
```suggestion
eip: 7557
```
Assigning next sequential EIP/ERC/RIP number.
Numbering changed to sequential from 7500 and is no longer the PR number.

Please also update the filename & Eth Magicians post

File: EIPS/eip-7557.md [hunk 2]
```diff
@@ -0,0 +1,370 @@
...
+
+This value is equivalent to the calldata cost of including two bytes long identifier of entries in `block_access_list`.
+
+### Calculating a reimbursement of the charged priority fee
+
+Each transaction pays an individual `priorityFeePerGas` value and redistributing this part of the cold access cost
+is more complex.\
+We propose the following approach to a fair reimbursement of the paid `priorityFee`:
+
+1. The validator gets paid the `priorityFee` for each cold access only once, but according to the highest `priorityFee`
```
Comment (g11tech) [inline EIPS/eip-7557.md:168]:
```suggestion
1. The builder gets paid the `priorityFee` for each cold access only once, but according to the highest `priorityFee`
```

File: EIPS/eip-7557.md [hunk 3]
```diff
@@ -0,0 +1,370 @@
...
+   among the transactions containing the said cold access.
+2. The rest of the Ether that was charged by the validator as a  `priorityFee` is redistributed back to all the senders
+   of transactions containing the same cold access in proportion to their **marginal contribution** to the total reimbursement.
+3. The `marginal contribution` of a transaction to the total reimbursement is defined as the difference between the sum total
```
Comment (g11tech) [inline EIPS/eip-7557.md:172]: so the contribution of the the most expensive transaction is the second largest priority fee * gas cost. although this is fine but cost could just be proportionately reimbursed keeping it simple

File: EIPS/eip-7557.md [hunk 4]
```diff
@@ -0,0 +1,370 @@
...
+With the block-level warming there is a change that makes it possible for a transaction sender to construct
+transactions with a large `accessList` that cost very little to be included, and this can be used to bloat the
+blockchain size.
+
+This potential bloating of the block size also presents a challenge for the propagation of the block in a P2P network.
+
+In order to minimize the cost of permanently storing access lists, we propose the following changes to
+the `execution_payload` structure:
+
+1. Add a new `block_access_list` field. \
```
Comment (g11tech) [inline EIPS/eip-7557.md:202]: block_access_list also could be a derived thing from the per tx access list

File: EIPS/eip-7557.md [hunk 5]
```diff
@@ -0,0 +1,370 @@
...
+A sample JSON representation of the data structure that represents such a structure and is used in the pseudocode below:
+
+```json
+[
+  {
+    "address": "0xde0b295669a9fd93d5f28d9ec85e40f4cb697bae",
+    "accessors": [
+      {
+        "sender": "0x90F8bf6A479f320ead074411a4B0e7944Ea8c9C1",
+        "priorityFeePerGas": "1000"
```
Comment (g11tech) [inline EIPS/eip-7557.md:81]: although priorityFeePerGas is important to derive the reimbursements but they are again a picked from the txs, also sender info is not really important at this level. the compressed per tx reference can again resolve this

**Pull request closed without merging**